cms/djangoapps/contentstore/views/entrance_exam.py

```diff
@@ -0,0 +1,206 @@
+    POST
+        Adds an entrance exam module to the specified course.
+    DELETE
+        Removes the entrance exam from the course
+    """
+    course_key = CourseKey.from_string(course_key_string)
+
+    # Deny access if the user is valid, but they lack the proper object access privileges
+    if not has_course_author_access(request.user, course_key):
+        return HttpResponse(status=403)
+
+    # Retrieve the entrance exam module for the specified course (returns 404 if none found)
+    if request.method == 'GET':
+        return _get_entrance_exam(request, course_key)
+
+    # Create a new entrance exam for the specified course (returns 201 if created)
+    elif request.method == 'POST':
+        response_format = request.REQUEST.get('format', 'html')
+        http_accept = request.META.get('http_accept')
+        if response_format == 'json' or 'application/json' in http_accept:
+            ee_min_score = request.POST.get('entrance_exam_minimum_score_pct', None)
+
+            # if request contains empty value or none then save the default one.
+            entrance_exam_minimum_score_pct = float(settings.ENTRANCE_EXAM_MIN_SCORE_PCT)
+            if ee_min_score != '' and ee_min_score is not None:
+                entrance_exam_minimum_score_pct = float(ee_min_score)
+            return create_entrance_exam(request, course_key, entrance_exam_minimum_score_pct)
+        return HttpResponse(status=400)
+
+    # Remove the entrance exam module for the specified course (returns 204 regardless of existence)
+    elif request.method == 'DELETE':
+        return delete_entrance_exam(request, course_key)
+
+    # No other HTTP verbs/methods are supported at this time
+    else:
+        return HttpResponse(status=405)
+
+
+def create_entrance_exam(request, course_key, entrance_exam_minimum_score_pct):
+    """
+    api method to create an entrance exam.
+    First clean out any old entrance exams.
+    """
+    _delete_entrance_exam(request, course_key)
+    return _create_entrance_exam(request=request, course_key=course_key, entrance_exam_minimum_score_pct=entrance_exam_minimum_score_pct)
+
+
+def _create_entrance_exam(request, course_key, entrance_exam_minimum_score_pct=50.0):
+    """
+    Internal workflow operation to create an entrance exam
+    """
+
+    # Confirm the course exists
+    course = modulestore().get_course(course_key)
+    if course is None:
+        return HttpResponse(status=400)
+
+    # Create the entrance exam item (currently it's just a chapter)
+    payload = {
+        'category': "chapter",
+        'display_name': "Entrance Exam",
+        'parent_locator': unicode(course.location),
+        'is_entrance_exam': True,
+    }
+    factory = RequestFactory()
+    internal_request = factory.post('/', json.dumps(payload), content_type="application/json")
+    internal_request.user = request.user
+    created_item = json.loads(create_item(internal_request).content)
```

Comment:
I mentioned this over hipchat, but I would have preferred to extract the data-only portions of this (perhaps into an `api.py` module), and then call that directly, rather than manufacturing a fake request.